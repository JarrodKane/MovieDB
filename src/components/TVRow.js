import React from "react";

const TVRow = props => {
  const {
    id,
    name,
    image,
    year,
    rate,
    lang,
    handleClickAdd,
    isSignedIn,
    watchList
  } = props;
  const percentageRating = rate * 10;
  const Poster = `https://image.tmdb.org/t/p/w500/${image}`;

  let addOrDis;
  if (isSignedIn === false) {
    addOrDis = (
      <td id={id} className=" center aligned negative" onClick={handleClickAdd}>
        <i id={id} aria-hidden="true" className="grey sign-in huge icon"></i>
        <div>Sign In</div>
      </td>
    );
  } else {
    // This is for before there is an array of watchlists returned
    if (watchList.length === 0) {
      addOrDis = (
        <td
          id={`A-${id}`}
          className="selectable center aligned  huge icon "
          onClick={handleClickAdd}
        >
          <i
            id={`A-${id}`}
            aria-hidden="true"
            className="grey heart huge icon"
          ></i>
        </td>
      );
      // This is for if there is an array returned but there are no movies in the watchlist yet
    } else if (watchList.results.length === 0) {
      console.log(watchList);
      addOrDis = (
        <td
          id={`A-${id}`}
          className="selectable center aligned  huge icon "
          onClick={handleClickAdd}
        >
          <i
            id={`A-${id}`}
            aria-hidden="true"
            className="grey heart huge icon"
          ></i>
        </td>
      );
    } else {
      let watchListS = watchList.results;
      for (let i = 0; i < watchListS.length; i++) {
        if (watchListS[i].id === id) {
          addOrDis = (
            <td
              id={`R-${id}`}
              className="selectable center aligned  huge icon "
              onClick={handleClickAdd}
            >
              <i
                id={`R-${id}`}
                aria-hidden="true"
                className="red heart huge icon"
              ></i>
            </td>
          );
          break;
        } else {
          addOrDis = (
            <td
              id={`A-${id}`}
              className="selectable center aligned  huge icon "
              onClick={handleClickAdd}
            >
              <i
                id={`A-${id}`}
                aria-hidden="true"
                className="grey heart huge icon"
              ></i>
            </td>
          );
        }
      }
    }
  }

  return (
    <tr id={id} className="">
      <td className="">
        <img
          src={Poster}
          alt={`Poster of ${name}`}
          className="ui small image "
        />
      </td>
      <td className="ui medium header">{name}</td>
      <td className="">{year}</td>
      <td className="">{`${percentageRating}%`}</td>
      <td className="">{lang}</td>

      {addOrDis}
    </tr>
  );
};

export default TVRow;

//<div class="ui red label">Red</div>
