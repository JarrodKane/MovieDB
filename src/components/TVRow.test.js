import React from "react";
import { shallow } from "enzyme";
import TVRow from "./TVRow";
describe("TVRow", () => {
  it("should render correctly with a year prop, and a watchList prop", () => {
    const component = shallow(<TVRow year="20001" watchList="" />);

    expect(component).toMatchSnapshot();
  });
});
