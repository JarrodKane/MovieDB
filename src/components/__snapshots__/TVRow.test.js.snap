// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TVRow should render correctly with a year prop, and a watchList prop 1`] = `
<tr
  className=""
>
  <td
    className=""
  >
    <img
      alt="Poster of undefined"
      className="ui small image "
      src="https://image.tmdb.org/t/p/w500/undefined"
    />
  </td>
  <td
    className="ui medium header"
  />
  <td
    className=""
  >
    2000
  </td>
  <td
    className=""
  >
    NaN%
  </td>
  <td
    className=""
  />
  <td
    className="selectable center aligned  huge icon "
    id="A-undefined"
  >
    <i
      aria-hidden="true"
      className="grey heart huge icon"
      id="A-undefined"
    />
  </td>
</tr>
`;
